internal/access/access_key_test.go

```diff
@@ -33,7 +40,9 @@ func TestAccessKeys_SelfManagement(t *testing.T) {
 		_, err = CreateAccessKey(c, key)
 		assert.NilError(t, err)
 
-		err = DeleteAccessKey(c, key.ID, "")
+		r := rCtx // shallow copy
+		r.Authenticated.AccessKey = &models.AccessKey{}
+		err = DeleteAccessKey(r, key.ID, "")
```

Comment:
These test changes were necessary because previously this test did not set `Authenticated.AccessKey` which caused a panic now that we're checking `Authenticated.AccessKey`. We will never encounter that scenario in production because an `AccessKey` is required for this endpoint.